packages/rest/src/__tests__/unit/router/rebase-openapi-spec.unit.ts

```diff
@@ -0,0 +1,144 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {rebaseOpenApiSpec} from '../../../router';
+
+describe('rebaseOpenApiSpec', () => {
+  it('does not modify the OpenApiSpec if it does not have paths', async () => {
+    const spec = {
+      title: 'Greetings',
+      components: {
+        responses: {
+          Hello: {
+            description: 'Hello.',
+          },
+        },
+      },
+      tags: [{name: 'greeting', description: 'greetings'}],
+    };
+    const rebasedSpec = rebaseOpenApiSpec(spec, '/api');
+
+    expect(spec).to.eql(rebasedSpec);
+  });
+
+  it('does not modify the OpenApiSpec if basePath is empty or `/`', async () => {
+    const spec = {
+      paths: {
+        '/': {
+          get: {
+            responses: {
+              '200': {
+                description: 'greeting',
+                content: {
+                  'application/json': {
+                    schema: {type: 'string'},
+                  },
+                },
+              },
+            },
+          },
+        },
+        '/hello': {
+          post: {
+            summary: 'says hello',
+            consumes: 'application/json',
+            responses: {
+              '200': {
+                description: 'OK',
+              },
+            },
+          },
+        },
+      },
+    };
+
+    let rebasedSpec = rebaseOpenApiSpec(spec, '');
+    expect(spec).to.eql(rebasedSpec);
```

Comment:
nitpick (we usually expect the result to match some expected value):
```suggestion
    expect(rebasedSpec).to.eql(spec);
```